### ISSUE TITLE
emojisums only displayable on GitHub due to non-Unicode emoji

### ISSUE THREAD

**Issue by wgwoods** [emojisum/emojisum#5]
## tl;dr

* `:finnadie:`, `:hurtrealbad:`, and `:trollface:` won't work anywhere except GitHub
* some of the other (newer) emoji may not be available on some devices
* bummer :frowning_face: 

## long version

...so, I was gonna port emojicode to Rust, for funsies. Except it turns out that the Rust [emojicons] crate didn't know about some of the items in `emojimap.json`! Specifically, it doesn't know what to do with these:

`cn` `de` `es` `fr` `fu` `eu` `it` `jp` `kr` `ru` `uk` `us` `metal` `finnadie` `hurtrealbad` `trollface`

As it turns out, these are three categories of missing items:

* ISO-3166 country codes
    * Easy enough to map to flags; something to fix in [emojicons], I guess.
* `metal`
    * This is an alias for :metal: U+1F918 "sign of the horns". [emojicons] doesn't know about those aliases, which I guess are defined in [gemoji]'s [`emoji.json`]. (again, that's something to fix upstream)
* `finnadie` `hurtrealbad` `trollface`
    * GitHub-specific magic for the poor abused Doom guy :finnadie: :hurtrealbad: and this jerk :trollface:. FIE ON THESE BOYS!

[emojicons]: https://crates.io/crates/emojicons
[gemoji]: https://github.com/github/gemoji
[`emoji.json`]: https://github.com/github/gemoji/blob/master/db/emoji.json

## ridiculous top-of-head suggested solution

* Have different `emojimap` objects for different purposes!
* Prefix the emojisum with a codepoint from the [Emoticons unicode block] (U+1F6xy), where:
    * x is the map number [0-4] as defined below
    * y is a CRC-4 of the rest of the checksum because YO DAWG I HEARD YOU LIKE CHECKSUMS, SO WE PUT A CHECKSUM IN YOUR CHECKSUM SO YOU CAN VERIFY WHILE YOU VERIFY

[Emoticons unicode block]: https://en.wikipedia.org/wiki/Emoticons_(Unicode_block)

Suggested maps:

0. `"version": "draft"`: uses the current emojiwords map as-is.
1. `"version": "unicode"`: minimally modify map 0, replacing GitHub-specific emoji.
2. `"version": "console"`: a map that uses symbols that are available in typical black-and-white console/wingdings fonts (ideally only items from Unicode 6.0 and older)
3. [reserved for future expansion]
4. [reserved for future expansion]

## less ridiculous solution

* Just replace the three offending emoji with Unicode-defined emoji
    * I mean, I _guess_ that works
    * yawwwwwwn tho

## in conclusion

EMOJISUM EVERYWHERE. LET'S MAKE IT HAPPEN.

**Comment by vbatts** [comment]
First off: SRS LOVE HERE
Moving on:
Fixing the flags should be straight forward. And even emoji that have synonyms. Perhaps the `emojimap.json` doesn't have to be a strict array of strings, but each item could be a map, that has the string or alternates, and whether there is a unicode designation for the emoji. Then for implementations that need to depend on the unicode can just.do.that.

The idea of U+ prefix is pretty cute. Would accomplish similar as above. except _efficiency_ vs *JSON*. One difference would be that with having a list of maps, each entry would still be correlated by it's position in the array == 0-254 byte, and not need alternate emojimaps.

Side note: just hacked out a golang CRC-4 implementation. But wasn't clear whether it needs to just be a `unit8` or not.

**Comment by vbatts** [comment]
Ok, with #6 landed, now each entry can have a unicode codepoint entry as well. So implementations that don't support the first emoji word, can check the next, etc.

Now to chunk away at the github only emojis, and adding the codepoints for flags etc.
Referencing https://github.com/jiri/rust-emojicons/blob/master/build.rs will be pretty straight forward for getting codepoints. 

Also, part of #6 was adding a simple markdown review of the current emojimap.json set. 
https://github.com/vbatts/emojisum/blob/master/emoji/list.md

**Comment by vbatts** [comment]
other possible alternatives could be `U+1F3A0` :carousel_horse: `:carousel_horse:`

**Comment by vbatts** [comment]
also, gonna slate :frog: to get the axe. because :middle_finger:

**Comment by vbatts** [comment]
looking at switch :frog: to :peanuts: (`U+1F95C`)

**Comment by vbatts** [comment]
perhaps switch :notes: to :control_knobs: (`:control_knobs:` `U+1F39B`), as notes looks like music score

**Comment by vbatts** [comment]
@wgwoods is this better now?

**Comment by vbatts** [comment]
on closer look, the rust emojicons is missing (from our current set):
```json
    [":cn:", "U+1F1E8", "U+1F1F3"],
    [":control_knobs:", "U+1F39B"],
    [":de:", "U+1F1E9", "U+1F1EA"],
    [":es:", "U+1F1EA", "U+1F1F8"],
    [":eu:", "U+1F1EA", "U+1F1FA"],
    [":female_detective:", "U+1F575", "U+1FE0F", "U+200D", "U+2640"],
    [":fr:", "U+1F1EB", "U+1F1F7"],
    [":fu:", "U+1F595"],
    [":heavy_heart_exclamation:", "U+2763"],
    [":it:", "U+1F1EE", "U+1F1F9"],
    [":jp:", "U+1F1EF", "U+1F1F5"],
    [":kr:", "U+1F1F0", "U+1F1F7"],
    [":metal:", "U+1F918"],
    [":nauseated_face:", "U+1F922"],
    [":peanuts:", "U+1F95C"],
    [":ru:", "U+1F1F7", "U+1F1FA"],
    [":uk:"],
    [":us:"],
```

**Comment by vbatts** [comment]
Not only is this done now! But there is also a rust implementation in the works (#23 thanks to @ goldenMetteyya )
